tools/misc/convert_mapping_yaml_to_excel.py

```
@@ -0,0 +1,248 @@
+            for idx, s in enumerate(sets[:2]):
+                if "requirement_mappings" not in s:
+                    print(f"❌ [ERROR] Missing 'requirement_mappings' in mapping set {idx}.")
+                    sys.exit(1)
+            return sets, 3
+        else:
+            idx = int(choice) - 1
+            if idx >= len(sets):
+                print(f"❌ [ERROR] Mapping index {idx} not found in 'requirement_mapping_sets'.")
+                sys.exit(1)
+            if "requirement_mappings" not in sets[idx]:
+                print(f"❌ [ERROR] Missing 'requirement_mappings' in selected mapping set.")
```

Ruff (0.11.9)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)